### PR TITLE
PDF Metadata

### PR DESCRIPTION
**Pull request by pmbean6**
This is a functionality addition. It uses the hyperref package (if loaded in the document) to automatically enter the author and title into the PDF metadata.
In addition, the \pdfsupresstexinfo call suppresses the gibberish information input by the pdflatex compiler

### CODE REVIEW

File: maine-thesis.cls [hunk 1]
```diff
@@ -79,6 +79,18 @@
 \RequirePackage[none]{hyphenat}[2009/09/02 v2.3c hyphenation utilities]
 \AfterPackage{hyperref}{% Graduate School requirements mandate a slight modification to hyperref's redefinition of the caption command.  This is implemented here.  The command is only executed after hyperref is loaded in the document (if it is loaded).
 \long\def\@caption#1[#2]#3{%
+\hypersetup{ pdfdisplaydoctitle=true}
```
Comment (rpspringuel) [inline maine-thesis.cls]: You've put this code block inside the redefined caption command, which will lead to it being executed at the wrong time (and multiple times).  It should be moved to be outside the caption command definition.

File: maine-thesis.cls [hunk 2]
```diff
@@ -79,6 +79,18 @@
...
+
+\AtBeginDocument{
+\hypersetup{	
+	pdftitle=\@title,
+	pdfauthor=\@author,
+	pdfcreator=\relax,
+	pdfproducer=\relax % These could be changed, but for the thesis, I left them blank.
+	}
+}
+\pdfsuppressptexinfo=-1 %to remove some useless tags generated by pdftex
```
Comment (rpspringuel) [inline maine-thesis.cls]: This command is specific to PDFTeX and not everyone uses that engine (I usually use LuaTeX these days, for instance).  Please add the `iftex` package to package loading and then enclose this line in the `\ifPDFTeX` conditional.

**Review by rpspringuel — CHANGES_REQUESTED** [review]

**Comment by pmbean6** [comment]
I am closing this pull request and creating a new one with just the hypersetup calls. 

When I have time to correct the suppresstexinfo call, I will make it a new edit

File: maine-thesis.cls (hunk 1 above)
Comment (pmbean6) [reply maine-thesis.cls]: Fixed. accidentally used 2 commits, and can't see how to alter that. I can close this pull, and open another if necessary.

File: maine-thesis.cls (hunk 2 above)
Comment (pmbean6) [reply maine-thesis.cls]: I used the iftex package to fix this as you suggested

File: maine-thesis.cls (hunk 1 above)
Comment (pmbean6) [reply maine-thesis.cls]: This was actually a typo, 
 in my own edited copy of the class file these calls were in the correct place originally. Glad you spotted that.

**Review by rpspringuel — APPROVED** [review]

**Comment by rpspringuel** [comment]
Looks good and now works for me just fine.

**Pull request merged**